### ISSUE TITLE
Validator broken

### ISSUE THREAD

**Issue by t-b** [NeurodataWithoutBorders/pynwb#715]
Take one of the docs/gallery/general nwb files and call `python -m pynwb.validate basic_example.nwb` and you get

```
$ python -m pynwb.validate basic_example.nwb
C:\Anaconda2\envs\py36\lib\site-packages\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to
 `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Traceback (most recent call last):
  File "C:\Anaconda2\envs\py36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Anaconda2\envs\py36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "e:\projekte\pynwb\src\pynwb\validate.py", line 58, in <module>
    main()
  File "e:\projekte\pynwb\src\pynwb\validate.py", line 52, in main
    errors = validate(io)
  File "e:\projekte\pynwb\src\pynwb\form\utils.py", line 390, in func_call
    return func(**parsed['args'])
  File "e:\projekte\pynwb\src\pynwb\__init__.py", line 186, in validate
    return validator.validate(builder)
  File "e:\projekte\pynwb\src\pynwb\form\utils.py", line 375, in func_call
    return func(self, **parsed['args'])
  File "e:\projekte\pynwb\src\pynwb\form\validate\validator.py", line 314, in validate
    return validator.validate(builder)
  File "e:\projekte\pynwb\src\pynwb\form\utils.py", line 375, in func_call
    return func(self, **parsed['args'])
  File "e:\projekte\pynwb\src\pynwb\form\validate\validator.py", line 391, in validate
    ret.extend(sub_val.validate(bldr))
  File "e:\projekte\pynwb\src\pynwb\form\utils.py", line 375, in func_call
    return func(self, **parsed['args'])
  File "e:\projekte\pynwb\src\pynwb\form\validate\validator.py", line 369, in validate
    ret = super(GroupValidator, self).validate(builder)
  File "e:\projekte\pynwb\src\pynwb\form\utils.py", line 375, in func_call
    return func(self, **parsed['args'])
  File "e:\projekte\pynwb\src\pynwb\form\validate\validator.py", line 221, in validate
    errors = validator.validate(attr_val)
  File "e:\projekte\pynwb\src\pynwb\form\utils.py", line 375, in func_call
    return func(self, **parsed['args'])
  File "e:\projekte\pynwb\src\pynwb\form\validate\validator.py", line 190, in validate
    if not check_type(spec.dtype, dtype):
  File "e:\projekte\pynwb\src\pynwb\form\validate\validator.py", line 78, in check_type
    received in 'ascii'
TypeError: 'in <string>' requires string as left operand, not numpy.dtype
(py36)
```

After this is fixed the validation tool should also be run as part of building the gallery files I guess.

**Comment by pgleeson** [comment]
Would be great to get this fixed, as well as having more simple example NWB files (and scripts to generate them) which are compliant to best practices.

**Comment by bendichter** [comment]
@t-b it looks like you've been making some great progress fixing the validator in #747, but I see that you removed yourself from this issue a week ago. Are you still working on the validator?

**Comment by bendichter** [comment]
@pgleeson It's the validator that is broken. To the best of my knowledge the tutorials do give examples of best practices.

**Comment by bendichter** [comment]
I changed the name because the validator is failing even on the simplest files. e.g.

```python
from datetime import datetime
from dateutil.tz import tzlocal
from pynwb import NWBFile, NWBHDF5IO

start_time = datetime(2017, 4, 3, 11, tzinfo=tzlocal())
nwbfile = NWBFile('demonstrate NWBFile basics', 'NWB123', start_time)

io = NWBHDF5IO('minimal_example.nwb', mode='w')
io.write(nwbfile)
io.close()
```

**Comment by t-b** [comment]
@bendichter Not at the moment. The next validation bugs I found are currently beyond my available time.

**Comment by bendichter** [comment]
@t-b Ok thanks for letting us know

**Comment by bendichter** [comment]
@ajtritt could you take a look at this?

**Comment by ajtritt** [comment]
fixed by #775
